tools/find_tap_name/main.go

```diff
@@ -0,0 +1,143 @@
+// hardware component ID does not match `componentID`.
+func getAdapterNameAndInstallTimestamp(adapterKeyPath, componentID string) (name string, installTimestamp uint64, err error) {
+  adapterKey, err := registry.OpenKey(registry.LOCAL_MACHINE, adapterKeyPath, registry.READ)
+  if err != nil {
+    log.Println("Failed to open adapter key:", err)
+    return
+  }
+  defer adapterKey.Close()
+
+  adapterComponentID, _, err := adapterKey.GetStringValue("ComponentId")
+  if err != nil {
+    log.Println("Failed to read adapter component ID:", err)
+    return
+  }
+  log.Println("Found", adapterComponentID)
+  if adapterComponentID != componentID {
+    err = fmt.Errorf("Network adapter component ID does not match %v", componentID)
+    return
+  }
+
+  installTimestampBytes, _, err := adapterKey.GetBinaryValue("InstallTimeStamp")
+  if err != nil {
+    log.Println("Failed to read adapter install timestamp:", err)
+    return
+  }
+  // Although Windows is little endian, we have observed that network adapters' install timestamps
+  // are encoded as big endian in the registry.
+  installTimestamp = binary.BigEndian.Uint64(installTimestampBytes)
+  log.Println("\tInstall timestamp", installTimestamp)
+
+  netConfigID, _, err := adapterKey.GetStringValue("NetCfgInstanceId")
+  if err != nil {
+    log.Println("Failed to read network configuration ID:", err)
+    return
+  }
+  adapterConfigKeyPath := fmt.Sprintf("%s\\%s\\Connection", netConfigKeyPath, netConfigID)
+  adapterConfigKey, err := registry.OpenKey(registry.LOCAL_MACHINE, adapterConfigKeyPath, registry.READ)
+  if err != nil {
+    log.Println("Failed to open network configuration key:", err)
+    return
+  }
+  defer adapterConfigKey.Close()
+
+  name, _, err = adapterConfigKey.GetStringValue("Name")
+  if err != nil {
+    log.Println("Failed to read adapter name:", err)
+    return
+  }
+  log.Println("\tName", name)
+  return
+}
+
+// findNetworkAdapterName searches the Windows registry for the name of a network adapter with
+// `componentID`. Since there may be more than one network adapter with the same component ID,
+// selects the most recently installed device in the event of a conflict.
+// Returns an empty string and an error if the device name cannot be found.
+func findNetworkAdapterName(componentID string) (string, error) {
+  netAdaptersKey, err := registry.OpenKey(registry.LOCAL_MACHINE, netAdaptersKeyPath, registry.READ)
+  if err != nil {
+    return "", fmt.Errorf("Failed to open the network adapter registry, %v", err)
+  }
+  defer netAdaptersKey.Close()
+
+  // List all network adapters.
+  adapterKeys, err := netAdaptersKey.ReadSubKeyNames(-1)
+  if err != nil {
+    return "", err
+  }
+
+  // Keep track of the most recently installed adapter name.
+  var name string
+  var installTimestamp uint64
+
+  for _, k := range adapterKeys {
+    adapterKeyPath := netAdaptersKeyPath + "\\" + k
```

Comment:
Nit: Can you use a backtick string instead of two backslashes (also for `adapterConfigKeyPath`)?  You also might want to consider the `path/filepath` package, or `strings.Join`.

Reply:
Done - went with `fmt.Sprintf`